addons/project/models/project.py

```diff
@@ -1366,9 +1367,43 @@ def _search_is_private(self, operator, value):
         else:
             return [('project_id', '!=', False)]
 
-    @api.depends('stage_id', 'project_id')
-    def _compute_kanban_state(self):
-        self.kanban_state = 'normal'
+    @api.depends('depend_on_ids.state', 'project_id.allow_task_dependencies')
+    def _compute_state(self):
+        for task in self:
+            if task.allow_task_dependencies:
+                for dependent_task in task.depend_on_ids:
+                    # if one of the blocking task is in a blocking state
+                    if dependent_task.state not in CLOSED_STATES:
+                        # here we check that the blocked task is not alread in a closed state (if the task is already done we don't put it in waiting state)
+                        if task.state not in CLOSED_STATES:
+                            task.state = '04_waiting_normal'
+                        return
```

Comment:
If i break there i'll just get out of the last loop and end up in the `if task.state == '04_waiting_normal'` condition. So i changed my code to avoid nested loops by using a list comprehension